go/pkg/sysdb/coordinator/coordinator.go

```diff
@@ -14,17 +14,29 @@ import (
 	"gorm.io/gorm"
 )
 
+// DeleteMode represents whether to perform a soft or hard delete
+type DeleteMode int
+
+const (
+	// SoftDelete marks records as deleted but keeps them in the database
+	SoftDelete DeleteMode = iota
+	// HardDelete permanently removes records from the database
+	HardDelete
+)
+
```

Comment:
My knowledge of go is rusty, but will this number from 0?  What happens if we want to extend this enum in the future?  I like that SoftDelete is the default, but worry about future proofing.